create.go

```diff
@@ -1,24 +1,20 @@
 package vaultrole
 
 import (
-	"strings"
-
 	"github.com/giantswarm/microerror"
-
-	"github.com/giantswarm/vaultrole/key"
 )
 
 func (r *VaultRole) Create(config CreateConfig) error {
-	k := key.WriteRolePath(config.ID, config.Organizations)
-	v := map[string]interface{}{
-		"allow_bare_domains": config.AllowBareDomains,
-		"allow_subdomains":   config.AllowSubdomains,
-		"allowed_domains":    key.AllowedDomains(config.ID, r.commonNameFormat, config.AltNames),
-		"organization":       strings.Join(config.Organizations, ","),
-		"ttl":                config.TTL,
+	c := writeConfig{
+		AllowBareDomains: config.AllowBareDomains,
+		AllowSubdomains:  config.AllowSubdomains,
+		AltNames:         config.AltNames,
+		ID:               config.ID,
+		Organizations:    config.Organizations,
+		TTL:              config.TTL,
 	}
 
-	_, err := r.vaultClient.Logical().Write(k, v)
+	err := r.write(c)
```

Comment:
Vault only provides a write operation. I want the library to provide create and update interfaces even though the implementation is the same.